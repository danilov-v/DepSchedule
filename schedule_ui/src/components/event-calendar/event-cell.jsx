import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

export function EventCell({ onClick, hasEvent }) {
  return (
    <div
      className={classnames("event-cell", { "event-cell-disabled": hasEvent })}
      onClick={onClick}
    />
  );
}

EventCell.propTypes = {
  onClick: PropTypes.func,
};
