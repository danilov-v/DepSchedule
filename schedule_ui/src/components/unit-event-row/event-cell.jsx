import React from "react";
import PropTypes from "prop-types";

import "./unit-event-row.scss";

export function EventCell({ eventDate, eventOwnerId, onClick }) {
  return (
    <div
      id={eventOwnerId}
      className="event-cell"
      onClick={() => {
        onClick && onClick(eventOwnerId, eventDate);
      }}
    ></div>
  );
}

EventCell.propTypes = {
  eventDate: PropTypes.instanceOf(Date),
  eventOwnerId: PropTypes.number,
  onClick: PropTypes.func,
};
