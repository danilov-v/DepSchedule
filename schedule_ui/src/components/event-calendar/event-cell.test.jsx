import React from "react";
import { shallow } from "enzyme";
import { EventCell } from "./event-cell";

describe("event-cell-row.jsx", () => {
  describe("<EventCell />", () => {
    it("should render and match snapshot", () => {
      const component = shallow(<EventCell />);

      expect(component).toMatchSnapshot();
    });

    it("should call onClick prop", () => {
      const mockFunction = jest.fn();
      const component = shallow(<EventCell onClick={mockFunction} />);

      component.find("div").simulate("click");
      expect(mockFunction).toHaveBeenCalled();
    });
  });
});
