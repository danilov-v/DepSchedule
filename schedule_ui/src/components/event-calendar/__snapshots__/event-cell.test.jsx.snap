// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event-cell-row.jsx <EventCell /> should render and match snapshot 1`] = `
<div
  className="event-cell"
/>
`;
